### PR TITLE
Added parts to pyo3 docu and added GPi, GPi2 gates

### PR DESCRIPTION
**Pull request by kbarkhqs**

### CODE REVIEW

File: roqoqo/src/operations/single_qubit_gate_operations.rs [hunk 1]
```diff
@@ -1566,3 +1566,191 @@ impl OperateSingleQubitGate for RotateXY {
...
+    fn beta_i(&self) -> CalculatorFloat {
+        (self.theta.clone()).cos() * -1.0 / 2.0_f64.sqrt()
+    }
+    /// Returns global_phase parameter of the operation.
+    ///
+    /// # Returns
+    ///
+    /// * `global_phase` - The global phase φ of the single-qubit unitary.
+    fn global_phase(&self) -> CalculatorFloat {
+        CalculatorFloat::from(0.0)
```
[the remark this answers is not in the document]
Comment (kbarkhqs) [reply roqoqo/src/operations/single_qubit_gate_operations.rs:1754]: Yes that's right, it's the only way I could get the unitary matrices based on alpha and beta to work out

File: roqoqo/tests/integration/operations/single_qubit_gate_operations.rs [hunk 2]
```diff
@@ -1040,7 +1107,8 @@ fn test_is_parametrized_false(gate: SingleQubitGateOperation) {
     CalculatorFloat::from(0),
     CalculatorFloat::from(0))); "Rotation")]
 #[test_case(SingleQubitGateOperation::from(PhaseShiftState0::new(0, CalculatorFloat::from(PI/2.0))); "phaseshiftstate0")]
-#[test_case(SingleQubitGateOperation::from(PhaseShiftState1::new(0, CalculatorFloat::from(PI/2.0))); "phaseshiftstate1")]
+#[test_case(SingleQubitGateOperation::from(GPi::new(0, CalculatorFloat::from(PI))); "gpi")]
```
Comment (mlodi-hqs) [inline roqoqo/tests/integration/operations/single_qubit_gate_operations.rs:1110]: PSS1 is gone?

File: .github/workflows/compatibility_tests.yml [hunk 3]
```diff
@@ -9,7 +9,7 @@ on:
 jobs:
 
   unittests_check_compatibility:
```
Comment (mlodi-hqs) [inline .github/workflows/compatibility_tests.yml:11]: just a small thing but we can follow the same structure and name this `unittest_check_compatibility_1_0`

File: .compatibility_tests/compatibility_test_1_2/tests/integration/compatibility_1_2.rs [hunk 4]
```diff
@@ -59,16 +41,54 @@ use test_roqoqo_1_2;
...
+#[test_case(test_roqoqo_1_2::operations::PragmaStopDecompositionBlock::new(vec![0]).into(); "PragmaStopDecompositionBlock")]
 #[test_case(test_roqoqo_1_2::operations::PragmaDamping::new(0,1.0.into(), 1.0.into()).into(); "PragmaDamping")]
+#[test_case(test_roqoqo_1_2::operations::PragmaDepolarising::new(0,1.0.into(), 1.0.into()).into(); "PragmaDepolarising")]
 #[test_case(test_roqoqo_1_2::operations::PragmaDephasing::new(0,1.0.into(), 1.0.into()).into(); "PragmaDephasing")]
 #[test_case(test_roqoqo_1_2::operations::PragmaRandomNoise::new(0,1.0.into(), 1.0.into(), 1.0.into()).into(); "PragmaRandomNoise")]
-#[test_case(test_roqoqo_1_2::operations::PragmaActiveReset::new(0).into(); "PragmaActiveReset")]
+#[test_case(test_roqoqo_1_2::operations::PragmaGeneralNoise::new(0, 0.1.into(), ndarray::array![[1.0.into(), 0.0.into()], [1.0.into(), 2.0.into()]]).into(); "PragmaGeneralNoise")]
 #[test_case(test_roqoqo_1_2::operations::PragmaConditional::new("ro".to_string(),0, test_roqoqo_1_2::Circuit::new()).into(); "PragmaConditional")]
+// #[test_case(test_roqoqo_1_2::operations::PragmaChangeDevice::new(&"PragmaTest".into()).unwrap().into(); "PragmaChangeDevice")]
```
Comment (mlodi-hqs) [inline .compatibility_tests/compatibility_test_1_2/tests/integration/compatibility_1_2.rs:62]: this stays out?

File: .compatibility_tests/compatibility_test_1_3/tests/integration/compatibility_1_3.rs [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+#[test_case(test_roqoqo_1_3::operations::PragmaStopDecompositionBlock::new(vec![0]).into(); "PragmaStopDecompositionBlock")]
+#[test_case(test_roqoqo_1_3::operations::PragmaDamping::new(0,1.0.into(), 1.0.into()).into(); "PragmaDamping")]
+#[test_case(test_roqoqo_1_3::operations::PragmaDepolarising::new(0,1.0.into(), 1.0.into()).into(); "PragmaDepolarising")]
+#[test_case(test_roqoqo_1_3::operations::PragmaDephasing::new(0,1.0.into(), 1.0.into()).into(); "PragmaDephasing")]
+#[test_case(test_roqoqo_1_3::operations::PragmaRandomNoise::new(0,1.0.into(), 1.0.into(), 1.0.into()).into(); "PragmaRandomNoise")]
+#[test_case(test_roqoqo_1_3::operations::PragmaGeneralNoise::new(0, 0.1.into(), ndarray::array![[1.0.into(), 0.0.into()], [1.0.into(), 2.0.into()]]).into(); "PragmaGeneralNoise")]
+#[test_case(test_roqoqo_1_3::operations::PragmaConditional::new("ro".to_string(),0, test_roqoqo_1_3::Circuit::new()).into(); "PragmaConditional")]
+// #[test_case(test_roqoqo_1_3::operations::PragmaChangeDevice::new(&"PragmaTest".into()).unwrap().into(); "PragmaChangeDevice")]
```
Comment (mlodi-hqs) [inline .compatibility_tests/compatibility_test_1_3/tests/integration/compatibility_1_3.rs:62]: here as well

**Review by nfwvogt — APPROVED** [review]

**Pull request merged**